### ISSUE TITLE
Strange errors during a tx call

### ISSUE THREAD

**Issue by chlenc** [FuelLabs/fuels-ts#2078]
### What version of fuels-ts are you using?

"fuels": "^0.77.0",

### Steps to Reproduce

clone spark matcher
```
git clone git@github.com:compolabs/spark-matcher.git 
cd spark-matcher
```

create .env fire with the next content
```
NODE_ENV=development
PORT=5003
PRIVATE_KEY=<YOUR PK WITH ETH>

MARKET="BTC"
```

```
 sudo docker build --tag spot-market-matcher-fuel-btc .
 sudo docker run --restart=always -d spot-market-matcher-fuel-btc
```

and check logs

### Expected Behavior

Works well like a few days ago

### Actual Behavior

Gm guys
I feel something is wrong today with a beta5, I am getting the errors like that randomly 
```
called `Result::unwrap()` on an `Err` value: SqueezedOutTransactionError("Transaction removed.")

called `Result::unwrap()` on an `Err` value: SqueezedOutTransactionError("Transaction expired because it exceeded the configured time to live `tx-pool-ttl`.")

error "Decode error: Custom { kind: Other, error: \"Response errors; Transaction is not inserted. UTXO is not existing: 0x4bfc230f928f6e23841026db7d1edbf9b926967e600805e035708a44e8406e7101\" }" })
```

for example, the spark matcher is stuck because of errors like

for orders 0x4d8e67903374b36c22fcf0fb07bd9a9060a9fbae37fd2c2a8f478335c9363ed8 
0x4285de56c7b84da6417f5bae151087b0dc8aa9b2746feed34d968d2d09fca62a 
```
 AssertFailedRevertError: The script reverted with reason AssertFailed
    0x72175cdd41bbf890f8cddfe54fa55ac0f311f963c010746337f1c2ac3d79ffbb: {"type":4,"val":"0xffffffffffff0004","pc":"0x13e94","is":"0x2da8"} 
```
the matcher works on ts, I tried to match orders in a rust sdk, and got something like
```
called `Result::unwrap()` on an `Err` value: IOError(Custom { kind: Other, error: "Decode error: Custom { kind: Other, error: \"Response errors; Transaction is not inserted. UTXO is not existing: 0xe2dfb444f5f0efceb44d827e515537d113645a1e8c5099115f7a2393ad50337d01\" }" })
```
or
```
called `Result::unwrap()` on an `Err` value: IOError(Custom { kind: Other, error: "Response errors; The specified coin(0xbaee48d07bac88ea56be391921b92abf1f64bb8d30df2395283aca1d745510db01) doesn't exist" })
```
After I tried to create the same orders locally (I got the data using a dry-run), it worked well
<img width="1028" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/35231749/041fac64-2648-4347-8ce3-d44d62b7f283">

**Comment by Torres-ssf** [comment]
Hey @chlenc, I am deeply sorry for this issue.

I have some questions to help us better understand the problem. 

> ### Expected Behavior
>
> Works well like a few days ago

You said this was working a few days ago. Were you using the same version when this was working? `0.77.0`?

> the matcher works on ts, I tried to match orders in a rust sdk, and got something like

Sorry, I did not understand this part. Is this working on the TS SDK but it is failing on the Rust SDK?

I've tried this on `beta-5` and I am having the same output.
![Screenshot 2024-04-17 at 10 38 38](https://github.com/FuelLabs/fuels-ts/assets/30977845/c1043663-2ccd-48ac-a3d2-dc2358c05a6d)

What is the contract function that is being called when I run `docker run --restart=always -d spot-market-matcher-fuel-btc` ?
